DOCS-1835: Update filtered-camera to use models from registry (#2537)
viamrobotics/docs: docs/tutorials/projects/filtered-camera.md

## The machine learning (ML) model for filtering

> Now that your camera component is steadily capturing images to the Viam app, you can train a machine learning (ML) model on those images to be able to detect certain objects in the camera feed.

Reviewer: I feel that somewhere we must still discuss why we use ML models, regardless of whether they are trained or uploaded or added (module):

> to be able to detect certain objects in the camera feed.

is still useful to me to understand what the next ~120 lines discuss.

Author: Added a suggestion of where I think it could fit

- [Detection](/ml/vision/#detections), where you use an ML model trained by drawing bounding boxes around distinct objects within captured images, which enables your machine to be able to detect those objects on its own.
- [Classification](/ml/vision/#classifications), where you use an ML model trained by tagging images with a class label that best describes it, which enables your machine to classify similar images on its own.

> In this tutorial, you will use filtering with a _detection_ model.

Author: ```suggestion
In this tutorial, you will use filtering with a _detection_ model to be able to detect certain objects in the camera feed and filter on the detected objects.
```

Reviewer: Perfect.

> To get a deteciton model you can train a machine learning (ML) model on images from your camera.

Reviewer: ```suggestion
To create a detection model, you can train a machine learning (ML) model on images captured by your camera.
```

For more information, see [Train a model](/ml/train-model/).

### Use an existing ML model

You can either [use an existing model from the registry](#select-an-existing-model-from-the-registry) or [upload an ML model](#upload-an-existing-ml-model) trained outside the Viam platform.

#### Select an existing model from the Registry

> The Viam registry hosts trained ML models that users have made public, which you can use to deploy classifiers or detectors for your use case onto your robot instead of training your own.

Reviewer: ```suggestion
The Viam registry provides pre-trained ML models that users have made public, which you can use to deploy classifiers or detectors for your use case onto your machine instead of training your own.
```

Nit: `host` suggest compute node, or webserver (active provenance of a service, rather than passive offering of a downloadable artifact). `provides` or `offers` or similar is more familiar for "resource can be found here"

`pre-trained` to match usage elsewhere on this page. Line 163 above walks us through uploading a not-pre-trained trained model, for example.

Author: I don't really understand what `pre-trained` is supposed to mean, there is no additional training possible (currently). And any model that gets uploaded is already trained? What does pre-trained mean to you?

Reviewer: Gotcha, I see the following two types:
- Pre-trained: Upload a model file & labels file, or add a module that provides an ML model. 
- Train yourself: Upload a bunch of photos, add labels, and train a new model on those photos.

Both are trained models, the former is done before this tutorial starts, the latter as part of this tutorial (so `pre-` helps to order operations in that regard).

The alternative would be to remove `trained` everywhere instead, and just call them `ML models` IMO:
- Upload an ML model, or add a module that provides an ML model.
- Train your own ML model.

But mostly: we use `pre-trained` elsewhere on this page, so should ideally match here as well. Across the docs, however, there are more instances of `trained` over `pre-trained`, so feel free to move this whole page to `trained` if you like, I could see that working 👍

You can see all available ML models in the [Viam Registry](https://app.viam.com/registry).

Once you've decided on a model to use, continue to [Add the ML model service](#add-the-ml-model-service).

> #### Upload an existing ML model

Reviewer: Line 209 below should likely change similarly to something like:

> To upload and use an existing ML model:

Or similar, as "use existing" is now a discrete separate action that does not require uploading (the new MR ML model approach)

Author: done
